components/UI/searchBar.tsx

```diff
@@ -1,68 +1,231 @@
+      zIndex: "1",
+      cursor: "pointer",
+    },
+    "&:hover fieldset": {
+      border: "1px solid #CDCCCC",
+    },
+    "& ::placeholder": {
+      color: "#B0AEAE",
+      textAlign: "center",
+      fontSize: "24px",
+      fontStyle: "normal",
+      fontWeight: "700",
+      lineHeight: "24px",
+      letterSpacing: "0.24px",
+      justifyContent: "center",
+      alignItems: "center",
+    },
+    "&.Mui-focused ::placeholder": {
+      color: "transparent",
+    },
+    "&.Mui-focused fieldset": {
+      borderColor: theme.palette.primary.main,
+    },
+  },
+}));
 
 type SearchBarProps = {
   onChangeTypedValue?: (typedValue: string) => void;
+  showHistory: boolean;
 };
+
 const SearchBar: FunctionComponent<SearchBarProps> = ({
   onChangeTypedValue,
+  showHistory,
 }) => {
   const router = useRouter();
+  const resultsRef = useRef<HTMLDivElement>(null);
   const [typedValue, setTypedValue] = useState<string>("");
   const isDomainValid = useIsValid(typedValue);
+  const [currentResult, setCurrentResult] = useState<SearchResult | null>();
+  const [searchResults, setSearchResults] = useState<SearchResult[]>([]);
+  const [showResults, setShowResults] = useState<boolean>(false);
+  const { provider } = useContext(StarknetIdJsContext);
+  const contract = new Contract(
+    naming_abi as Abi,
+    process.env.NEXT_PUBLIC_NAMING_CONTRACT as string,
+    provider as Provider
+  );
+
+  useEffect(() => {
+    const existingResults =
+      JSON.parse(localStorage.getItem("search-history") as string) || [];
+    const firstResults = existingResults.slice(0, 5);
+    const resultPromises = firstResults.map(
+      (result: { name: string; lastAccessed: number }) =>
+        getStatus(result.name, result.lastAccessed)
+    );
+    Promise.all(resultPromises).then((fullResults) => {
+      fullResults.sort(
+        (firstResult: SearchResult, secondResult: SearchResult) =>
+          secondResult.lastAccessed - firstResult.lastAccessed
+      );
+      setSearchResults(fullResults);
+    });
+  }, []);
+
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        resultsRef.current &&
+        !resultsRef.current.contains(event.target as Node)
+      ) {
+        setShowResults(false);
+      } else {
+        setShowResults(true);
+      }
+    }
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
 
   function handleChange(value: string) {
     setTypedValue(value.toLowerCase());
+    getStatus(value).then((result) => {
+      setCurrentResult(result);
+    });
   }
 
-  function onEnter(ev: KeyboardEvent<HTMLDivElement>) {
-    if (ev.key === "Enter") {
-      search(typedValue);
-      ev.preventDefault();
+  async function getStatus(
+    name: string,
+    lastAccessed?: number
+  ): Promise<SearchResult> {
+    const isDomainValid = useIsValid(name);
```

Comment:
I don't think you can use a hook here. May be convert it to a utils function to avoid the compiler complaints.